src/Npgsql/Replication/PgOutput/Messages/PrepareMessageBase.cs

```diff
@@ -0,0 +1,41 @@
+﻿using NpgsqlTypes;
+using System;
+
+namespace Npgsql.Replication.PgOutput.Messages;
+
+/// <summary>
+/// Abstract base class for the logical replication protocol begin prepare and prepare message
+/// </summary>
+public abstract class PrepareMessageBase : PreparedTransactionMessageBase
+{
+    /// <summary>
+    /// The LSN of the prepare.
+    /// </summary>
+    public NpgsqlLogSequenceNumber PrepareLsn => StartLsn;
+
+    /// <summary>
+    /// The end LSN of the prepared transaction.
+    /// </summary>
+    public NpgsqlLogSequenceNumber PrepareEndLsn => EndLsn;
+
+    /// <summary>
+    /// Prepare timestamp of the transaction.
+    /// </summary>
+    public DateTime TransactionPrepareTimestamp => Timestamp;
```

Comment:
Unrelated thought: we may want to document the kind of timestamps - I don't remember any more if it's UTC or not, would be good for the docs to be explicit on that (we'd ideally do this elsewhere too)

Reply:
I *think* it is UTC but I agree that I should look it up one last time and document it.

Reply:
It is even worse than that.
It is UTC (timestamp with time zone) but we return it as unspecified which as of Npgsql 6.0 we shouldn't in order to be consistent with our own API.

I'll open a separate issue for that because that affects all replication APIs returning `DateTime` so fixing it will be well outside the scope of this PR.

Reply:
:vomiting_face: 

Thanks for taking a look... Yeah, we should clean that up.